dracut/99setup-root/initrd-setup-root

```diff
@@ -86,6 +86,10 @@ systemd-tmpfiles --root=/sysroot --create \
     baselayout.conf baselayout-usr.conf \
     baselayout-home.conf
 
+# Give the correct ownernship otherwise the tmpfiles configuration will not
+# apply because of mismatching machine-id - which makes "journalctl --user" to fail.
+chown --recursive root:systemd-journal /run/log/journal/
```

Comment:
Wait, this reminds me of https://github.com/flatcar/Flatcar/issues/799
I think the problem could be that `systemd-tmpfiles` doesn't find the `systemd-journal` group because it's not part of `/sysroot/etc/group` but only loaded via the nss-altfiles plugin from our database in baselayout.
I think as long as we don't change systemd-tmpfiles to somehow resolve the nss stuff in the given chroot, we should all `systemd-journal` as special group to copy like `docker`: https://github.com/flatcar/baselayout/blob/flatcar-master/scripts/flatcar-tmpfiles#L13
This will also be a workaround for the linked issue (at least for the journal but not for `kvm` - maybe add that, too? At some point we should fix the tools, though).

Reply:
Based on the assumption that this existing tmpfiles rule here would set the group:
```
/usr/lib/tmpfiles.d/systemd.conf:Z /run/log/journal/%m ~2750 root systemd-journal - -
```

Reply:
Ah, ok, right, `/sysroot/run` isn't mounted and this setup would take place later and not in the call we do here.
If this is about `/run/log/journal` of the initrd I think it's at the wrong place.

Reply:
I think it's worth trying to enable it again and see if the ordering cycle is still there and how to resolve this. Moving things closer to upstream behavior can avoid other future problems.

Reply:
I think we can try `sed -i 's/After=local-fs.target/After=local-fs-pre.target/g' "$initdir/usr/lib/systemd/system/systemd-tmpfiles-setup.service"`

Reply:
I tried it out manually with `rd.shell rd.break=cmdline` (`=pre-mount` or `=pre-pivot` would be too late) on a qemu arm64 VM.
After the VM booted I see this:
```
core@localhost ~ $ ls -alh /run/log/
total 0
drwxr-xr-x.  3 root root             60 Mar 28 03:10 .
drwxr-xr-x. 30 root root            780 Mar 28 03:16 ..
drwxr-sr-x.  4 root systemd-journal  80 Mar 28 03:15 journal
core@localhost ~ $ ls -alh /run/log/journal/
total 0
drwxr-sr-x. 4 root systemd-journal 80 Mar 28 03:15 .
drwxr-xr-x. 3 root root            60 Mar 28 03:10 ..
drwxr-s---. 2 root systemd-journal 60 Mar 28 03:10 931d6b0b22a24775bbd156cfaf0e98f4
drwxr-s---. 2 root systemd-journal 60 Mar 28 03:15 bd9ce2b134744c6db1aca45238fd9848
core@localhost ~ $ ls -alh /run/log/journal/*
/run/log/journal/931d6b0b22a24775bbd156cfaf0e98f4:
total 2.5M
drwxr-s---. 2 root systemd-journal   60 Mar 28 03:10 .
drwxr-sr-x. 4 root systemd-journal   80 Mar 28 03:15 ..
-rw-r-----. 1 root systemd-journal 2.5M Mar 28 03:14 system.journal

/run/log/journal/bd9ce2b134744c6db1aca45238fd9848:
total 2.5M
drwxr-s---. 2 root systemd-journal   60 Mar 28 03:15 .
drwxr-sr-x. 4 root systemd-journal   80 Mar 28 03:15 ..
-rw-r-----. 1 root systemd-journal 2.5M Mar 28 03:16 system.journal
core@localhost ~ $ ls -alh /run/log/journal/*/*
-rw-r-----. 1 root systemd-journal 2.5M Mar 28 03:14 /run/log/journal/931d6b0b22a24775bbd156cfaf0e98f4/system.journal
-rw-r-----. 1 root systemd-journal 2.5M Mar 28 03:16 /run/log/journal/bd9ce2b134744c6db1aca45238fd9848/system.journal
```

Reply:
(I have no answer for why this happens only on arm64 and why the upstream file wants `local-fs` - probably more for the real system and not really needed in the initrd, but no idea how to encode this conditionally)

Reply:
It's needed because Ignition can repartition the rootfs (https://github.com/flatcar/bootengine/blob/flatcar-master/dracut/30ignition/ignition-disks.service#L24)